client/renderers/EGL/desktop_rects.c

```diff
@@ -0,0 +1,267 @@
+/**
+ * Looking Glass
+ * Copyright (C) 2017-2021 The Looking Glass Authors
+ * https://looking-glass.io
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "desktop_rects.h"
+#include "common/debug.h"
+#include "common/KVMFR.h"
+#include "common/locking.h"
+
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+#include <GLES3/gl3.h>
+
+#include "util.h"
+
+struct EGL_DesktopRects
+{
+  GLuint  buffers[2];
+  GLuint  vao;
+  int     count;
+  int     maxCount;
+};
+
+bool egl_desktop_rects_init(EGL_DesktopRects ** rects, int maxCount)
+{
+  *rects = malloc(sizeof(EGL_DesktopRects));
+  if (!*rects)
+  {
+    DEBUG_ERROR("Failed to malloc EGL_DesktopRects");
+    return false;
+  }
+  memset(*rects, 0, sizeof(EGL_DesktopRects));
+
+  glGenVertexArrays(1, &(*rects)->vao);
```

Comment:
I am trying to get this project away from this usage... `(*rects)->`, please use a local, ie:
```
bool egl_desktop_rects_init(EGL_DesktopRects ** rects_, int maxCount)
{
  EGL_DesktopRects * rects = malloc(sizeof(EGL_DesktopRects));
  *rects_ = rects;
  ...
}
```

Reply:
Done.